### PR TITLE
fix paddle.where torch diff

### PR DESCRIPTION
**Pull request by ronny1996**
### PR types

Others
### PR changes

APIs
### Describe

paddle.where(condition)行为对齐torch.where(condition)和np.where(condition)

image
image

image
image

文档PR：https://github.com/PaddlePaddle/docs/pull/4190

### CODE REVIEW

File: python/paddle/tensor/search.py [hunk 1]
```diff
@@ -523,9 +523,10 @@ def mode(x, axis=-1, keepdim=False, name=None):
     return values, indices
 
 
-def where(condition, x, y, name=None):
+def where(condition, x=None, y=None, name=None):
     r"""
-    Return a tensor of elements selected from either $x$ or $y$, depending on $condition$.
+    Return a tensor of elements selected from either $x$ or $y$, depending on $condition$. paddle.where(condition)
+    is identical to paddle.nonzero(condition, as_tuple=True), when x and y is None.
```
Comment (TCChenlong) [inline python/paddle/tensor/search.py]: paddle.where(condition) is identical to paddle.nonzero(condition, as_tuple=True), when x and y is None.

This sentence should be written in a note.
Please refer to: https://github.com/PaddlePaddle/docs/wiki/%E9%A3%9E%E6%A1%A8API%E6%96%87%E6%A1%A3%E4%B9%A6%E5%86%99%E8%A7%84%E8%8C%83#%E6%B3%A8%E8%A7%A3

File: python/paddle/tensor/search.py [hunk 2]
```diff
@@ -559,7 +560,19 @@ def where(condition, x, y, name=None):
 
           print(out)
           #out: [1.0, 1.0, 3.2, 1.2]
+
```
Comment (TCChenlong) [inline python/paddle/tensor/search.py:565]: L542 & L543:
x and y should be described more clearly.

**Review by TCChenlong — APPROVED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**